src/main/java/com/denizenscript/clientizen/scripts/containers/gui/GuiScriptContainer.java

```diff
@@ -0,0 +1,440 @@
+                right = getTaggedInt(config, "right", context);
+        if (top == null || left == null || bottom == null || right == null) {
+            Debug.echoError("Invalid insets: must have top/left/bottom/right values.");
+            return null;
+        }
+        return new Insets(top, left, bottom, right);
+    }
+
+    // <--[language]
+    // @name GUI Textures
+    // @group GUI System
+    // @description
+    // GUI Textures directly relate to textures loaded in from resource packs (including the built-in one).
+    // They can be a single texture, or a section of a texture from a sprite sheet.
+    // Since these use the resources system, you can also add and use custom textures yourself via custom resource packs.
+    //
+    // Textures can either be a single texture path: minecraft:textures/block/cobblestone.png
+    // Or a section of a sprite sheet:
+    // <code>
+    // # Required, the sprite sheet to use a part of
+    // texture: <texture path>
+    // # The left edge of the area to use
+    // u_start: <decimal>
+    // # The top edge of the area to use
+    // v_start: <decimal>
+    // # The right edge of the area to use
+    // u_end: <decimal>
+    // # The bottom edge of the area to use
+    // v_end: <decimal>
+    // </code>
+    // All of the UV values are between 0 and 1 and are relative to the entire image.
+    // So for example: 0, 0, 1, 1 (in order) will be the entire image, 0.5, 0.5, 1, 1 will be the bottom right half, etc.
+    // -->
+
+    public static Texture parseTexture(YamlConfiguration config, String path, TagContext context) {
+        YamlConfiguration textureConfig = config.getConfigurationSection(path);
+        String pathStr = textureConfig != null ? getTaggedString(textureConfig, "texture", context) : getTaggedString(config, path, context);
+        if (pathStr == null) {
+            if (textureConfig != null) {
+                Debug.echoError("Invalid texture: must specify a texture path.");
+            }
+            return null;
+        }
+        Identifier texturePath = Identifier.tryParse(pathStr);
+        if (texturePath == null) {
+            Debug.echoError("Invalid texture: invalid texture path '" + pathStr + "' specified.");
+            return null;
+        }
+        if (textureConfig == null) {
+            return new Texture(texturePath);
+        }
+        Texture.Type type = getTaggedEnum(Texture.Type.class, Texture.Type.STANDALONE, textureConfig, "type", context);
+        if (type == null) {
+            Debug.echoError("Invalid texture: invalid type specified.");
+            return null;
+        }
+        Float uStart = getTaggedFloat(textureConfig, "u_start", 0f, context),
+                vStart = getTaggedFloat(textureConfig, "v_start", 0f, context),
+                uEnd = getTaggedFloat(textureConfig, "u_end", 1f, context),
+                vEnd = getTaggedFloat(textureConfig, "v_end", 1f, context);
```

Comment:
Usually these are called `u`, `v`, `width`, and `height`

Reply:
How would that apply here? these aren't `width`/`height` values, they're `right`/`left`/`top`/`bottom` points - unless you're saying to make it a top left point and take in width and height values to make the bottom right point?

Reply:
Yes, you would have to change them to be that top left point thing you described, but u/v start and end values aren't very useful.. also, the same texture flipping phenomenon becomes clearer when  you have negative width/height values instead of just, "not-forward" u/v end values

Reply:
Should be done, not 100% sure about the meta for the `u` and `v` values - couldn't really find a nicer way to explain them being the top left corner.